restli-int-test/src/test/java/com/linkedin/restli/examples/documentation/TestCustomDocumentationHandler.java

```diff
@@ -0,0 +1,113 @@
+/*
+   Copyright (c) 2021 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.restli.examples.documentation;
+
+import com.linkedin.data.DataMap;
+import com.linkedin.data.schema.DataSchemaResolver;
+import com.linkedin.data.schema.NamedDataSchema;
+import com.linkedin.r2.RemoteInvocationException;
+import com.linkedin.restli.client.Request;
+import com.linkedin.restli.common.OptionsResponse;
+import com.linkedin.restli.docgen.DefaultDocumentationRequestHandler;
+import com.linkedin.restli.docgen.RestLiDocumentationRenderer;
+import com.linkedin.restli.docgen.RestLiJSONDocumentationRenderer;
+import com.linkedin.restli.docgen.RestLiResourceRelationship;
+import com.linkedin.restli.examples.RestLiIntegrationTest;
+import com.linkedin.restli.examples.greetings.client.GreetingsRequestBuilders;
+import com.linkedin.restli.server.RestLiConfig;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+/**
+ * Integration tests for configuring server with a custom JSON documentation handler..
+ *
+ * @author Karthik Balasubramanian
+ */
+public class TestCustomDocumentationHandler extends RestLiIntegrationTest
+{
+
+  public static final String CUSTOM_SUFFIX = ".Custom";
+
+  @BeforeClass
+  public void initClass() throws Exception
+  {
+    RestLiConfig config = new RestLiConfig();
+    config.setDocumentationRequestHandler(new DefaultDocumentationRequestHandler()
```

Comment:
I did not know we can override the DocumentationRequestHandler. It looks like I can even rewrite a new DefaultDocumentationRequestHandler without the change you made to the RestLiDocumentationRenderer interface, though it will be a lot of duplicated code.

Reply:
You could also extend the DefaultDocumentationrequestHandler and just override how you construct the HTML/JSON renderers. 
Basically the hooks are provided at multiple levels:
 - Use a custom documentation request handler.
 - Use the default handler, but change the HTML/JSON renderers
 - Use the provided renderers, but change how you handle data schemas.